### ISSUE TITLE
Print problem when importing this project

### ISSUE THREAD

**Issue by UserAEH** [Deci-AI/super-gradients#1021]
My problem is that no print is displayed in the command line when using this module:
A simple example could show this problem:
```
from super_gradients.training import models
print("hello")
```
but no print appears on the console.
I tried with sys.stdout.write("hello") and sys.stdout.flush() but no results.
Is there any conflict with default print function? Do you have any idea about how to solve this issue?

**Comment by Louis-Dupont** [comment]
Hi @UserAEH , can you try:

``` python
stdout = sys.stdout

from super_gradients.training import models

sys.stdout = stdout

print("hello")
```

**Comment by UserAEH** [comment]
> Hi @UserAEH , can you try:
> 
> ```python
> stdout = sys.stdout
> 
> from super_gradients.training import models
> 
> sys.stdout = stdout
> 
> print("hello")
> ```

It works perfectly, thanks.
Do you think there's some problem within the package or my environment? (In case there's a problem with the package, we could request a fix to enhance the project)

**Comment by Louis-Dupont** [comment]
It's the first time someone faced this issue, I guess there must be a conflict between SuperGradients and something else in your environment. Did you import anything before the snippet you showed, or it happened when just importing `super_gradients`?

**Comment by UserAEH** [comment]
No, there's nothing before the snippet. I posted the full code.
I'm working with Python 3.10.11 if you want to test it.

**Comment by Louis-Dupont** [comment]
Tried on 3.10.11 as well but it doesn't reproduce on my device. 
I will investigate when I have some time, in the meantime I guess the workaround is to wrap the first super-gradient import with it. I think that the next imports won't create this issue again.

You can also try the following code, it might work as well:

``` python
import sys
from super_gradients.training import models

sys.stdout = sys.__stdout__

print("hello")
```

**Comment by UserAEH** [comment]
> Tried on 3.10.11 as well but it doesn't reproduce on my device. I will investigate when I have some time, in the meantime I guess the workaround is to wrap the first super-gradient import with it. I think that the following imports won't create this issue.
> 
> You can also try the following code, it might work as well:
> 
> ```python
> import sys
> from super_gradients.training import models
> 
> sys.stdout = sys.__stdout__
> 
> print("hello")
> ```

This solution is working. It seems like the problem is coming from the file common/abstractions/mute_processes.py.
When i commented the line 30:
`#sys.stdout = open(os.devnull, "w")`

Print worked with the first code i posted and without reassigning original value to stdout.
Another thing to mention is that there was one line that was added in the console before printing hello which i think is normal:

> The console stream is logged into some\path\logs\console.log.

**Comment by UserAEH** [comment]
Hi @Louis-Dupont , i tried to install from master. The encoding is added but the issue persisted.

**Comment by ukoehler** [comment]
Problem confirmed for 
super-gradients               3.1.1
Platform Windows 10

from super_gradients.training import Trainer is enough to cause the problem for me.

Workaround sys.stdout = sys.__stdout__ works, but it is still annoying.

**Comment by pcismyname** [comment]
same problem here(Window 10)!

**Comment by xcweric** [comment]
I found the solution:
(super_gradients 3.1.3)
\common\abstractions\abstract_logger.py
commented the line 9&18: 'mute_subprocesses()'
Then log and print() will show again:)
by default this func is always called which not friendly for some users like me:( , I spent a lot of time to find it...

**Comment by UserAEH** [comment]
> I found the solution: (super_gradients 3.1.3) \common\abstractions\abstract_logger.py commented the line 9&18: 'mute_subprocesses()' Then log and print() will show again:) by default this func is always called which not friendly for some users like me:( , I spent a lot of time to find it...

The problem in your solution is that It prints also logger messages also so I am not sure if it is a normal behavior. Logger should be written in log file not printed in the CLI.

**Comment by cobaltautomationdev** [comment]
this issue still exists until now 2024.01.14
